packages/e2e-tests/specs/editor/various/inserting-blocks.test.js

```diff
@@ -304,10 +304,10 @@ describe( 'Inserting blocks', () => {
 			inserterMenuInputSelector
 		);
 		inserterMenuSearchInput.type( 'cover' );
-		await page.waitForSelector(
-			'.block-editor-block-types-list .editor-block-list-item-cover'
-		);
-		// clicking may be too quick and may select a detached node.
+		// Wait for the search results to load after typing the full search
+		// term. Previously this test would sometimes accidentally tab to the
+		// button for a code block.
+		await page.evaluate( () => new Promise( window.requestIdleCallback ) );
```

Comment:
I don't believe this check would necessarily run the scheduled event we are looking for (load search results) always.
This can still cause intermittent failures and it does in trunk.

I've changed this check here: https://github.com/WordPress/gutenberg/pull/31047/commits/b92843872fb63b9f4744c282ae63df4b2daca413 which will land now with a green CI.

Reply:
That's what we had before, which was also failing sometimes. But the failures happen more now. I'll just revert my PR.

Reply:
Reverted in 5a2a966ac49c843b0146a4a581173e6e0e4579ad.